admin-dev/themes/new-theme/js/pages/carrier/form/manager/carrier-form-manager.ts

```diff
@@ -0,0 +1,237 @@
+        ranges.forEach((range: Range, index) => {
+          // Then, we append the new range row into the range container
+          const $rPrototype = this.prepareRangePrototype(rangePrototype, index, range);
+          $rangeContainer.append($rPrototype);
+        });
+      }
+    });
+  }
+
+  private onDeleteZone(e: Event) {
+    e.preventDefault();
+
+    // We need to get the zone id to delete
+    const $currentTarget = $(e.currentTarget as HTMLElement);
+    const $currentZoneRow = $currentTarget.parents(CarrierFormMap.zoneRow);
+    const idZoneToDelete = $currentZoneRow.children('input[name$="[zoneId]"]').val();
+
+    // We need to display a confirmation modal before deleting the zone
+    const modal = new ConfirmModal(
+      {
+        id: 'modal-confirm-submit-feature-flag',
+        confirmButtonClass: 'btn-danger',
+        confirmTitle: $currentTarget.data('modal-title'),
+        confirmMessage: '',
+        confirmButtonLabel: $currentTarget.data('modal-confirm'),
+        closeButtonLabel: $currentTarget.data('modal-cancel'),
+      },
+      () => {
+        // If, the user confirms the deletion, we need to remove the zone
+        // First, we need to remove this zone from the zones
+        let zones = <string[]> this.$zonesInput.val() || [];
+        zones = zones.filter((zoneId: string) => zoneId !== idZoneToDelete);
+
+        // And update the zones selected values and trigger the zones selector change event
+        this.$zonesInput.val(zones);
+        this.$zonesInput.change();
+      },
+    );
+    modal.show();
+  }
+
+  private onChangeRanges(ranges: Range[]) {
+    // We retrieve all ranges containers in the page
+    const rangesContainer = $(CarrierFormMap.rangesContainer);
+
+    // For each range container, we need to update the ranges
+    rangesContainer.each((_, rangeContainer: HTMLElement) => {
+      // First, we need to save all values for this range.
+      const $rangeContainer = $(rangeContainer);
+      const pricesRanges = $(rangeContainer).find(CarrierFormMap.rangeRow).map((__, rangeRow: HTMLElement) => {
+        const $rangeRow = $(rangeRow);
+        const from = $rangeRow.find('input[name$="[from]"]').val();
+        const to = $rangeRow.find('input[name$="[to]"]').val();
+        const price = $rangeRow.find('input[name$="[price]"]').val();
+
+        return {from, to, price};
+      });
+
+      // Then, we reset the ranges container
+      $rangeContainer.html('');
+
+      // and, we need to add all the ranges selected
+      const rangePrototype = $rangeContainer.data('prototype');
+      ranges.forEach((range: Range, index) => {
+        // First, we need to prepare the range prototype
+        const $rPrototype = this.prepareRangePrototype(rangePrototype, index, range);
+        // We set the previous value for this range if it exists
+        // @ts-ignore
+        $rPrototype.find('input[name$="[price]"]').val(pricesRanges[index]?.price);
```

Comment:
Are you sure the index from the new ranges is always a match for the old ranges?

Shouldn't you try to match by identical from/to instead? Especially since `pricesRanges` contains from/to but you don't seem to use them for now